projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -0,0 +1,115 @@
+import { MediaUpload, MediaUploadCheck } from '@wordpress/block-editor';
+import { Button, ResponsiveWrapper, Spinner } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { useCallback, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import useAttachedMedia from '../../hooks/use-attached-media';
+import styles from './styles.module.scss';
+
+/**
+ * Get relevant details from a WordPress media object.
+ *
+ * @param {Object} media - WordPress media object.
+ * @returns {{ mediaWidth: [ number ], mediaHeight: [ number ], mediaSourceUrl: [ string ] }} - Media details.
+ */
+const getMediaDetails = media => {
+	if ( ! media ) {
+		return {};
+	}
+
+	const sizes = media?.media_details?.sizes ?? {};
+	const mediaObject = sizes.large || sizes.thumbnail || media.media_details;
+
+	return {
+		mediaWidth: mediaObject.width,
+		mediaHeight: mediaObject.height,
+		mediaSourceUrl: mediaObject.source_url,
+	};
+};
+
+/**
+ * Wrapper that handles media-related functionality.
+ *
+ * @returns {object} The media section.
+ */
+export default function MediaSection() {
+	const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
+	const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
+	const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
+
+	const ALLOWED_MEDIA_TYPES = useMemo( () => [ 'image/jpeg', 'image/png' ], [] );
```

Comment:
If we were to define these outside of `MediaSection` like `getMediaDetails` that should mean we can avoid having to memoize ALLOWED_MEDIA_TYPES, and subsequently adding these to hook dependency arrays.

Reply:
Good idea! Fixed.